### PR TITLE
Aria2: Edit config and update

### PR DESCRIPTION
**Pull request by kuoruan**
* Add more feature choices: Async DNS Support, Firefox3/Chromium Cookie Support.
* Switch default XML parser library to Libxml2, Expat has no update for a long time.
* Bump version to v1.31.0.
* Enable all features by default.

This commit https://github.com/openwrt/packages/commit/2b8aeb50029bb3a46517bd0508b05925bd793f0c revert the default setting, but I think we should enable them.

Maintainer: Hsing-Wang Liao <kuoruan@gmail.com> / @kuoruan
Compile tested: master version
Run tested: arm

### CODE REVIEW

**Comment by kuoruan** [comment]
@Ansuel https://github.com/openwrt/packages/pull/4028

File: net/aria2/Makefile [hunk 1]
```diff
@@ -52,19 +54,19 @@ endef
 CONFIGURE_ARGS += \
 	--disable-nls \
 	$(if $(CONFIG_ARIA2_NOSSL),--disable,--enable)-ssl \
+	$(if $(CONFIG_ARIA2_BITTORRENT),--enable,--disable)-bittorrent \
+        $(if $(CONFIG_ARIA2_METALINK),--enable,--disable)-metalink \
```
Comment (hnyman) [inline net/aria2/Makefile]: minor whitespace error. 
(other lines have properly tabs, while this has spaces?)

**Comment by Ansuel** [comment]
Good thx

File: net/aria2/Makefile (hunk 1 above)
Comment (kuoruan) [inline net/aria2/Makefile]: Sorry, fixed.

**Comment by hnyman** [comment]
Making all those default will add quite many dependencies to Aria. Buildbots would build the default, so everybody who downloads and installs Aria2, would need also all those packages that it will depend on.

What is the size increase of the installation? Aria2 size diff + the size of those additional dependencies?

It might be sensible to leave the basic package more minimal, and those who need a full-flavoured version can compile it by themselves.

**Comment by kuoruan** [comment]
@hnyman OK, reverted. pls check.

**Comment by ilfirin-ms** [comment]
The question is, why would anyone want to install aria2, when you cut all features off to reduce size? You have succeded, it's reduced to zero, because noone wants to install it now.

**Pull request merged**